common/src/main/java/discord4j/common/ReactorResourceProvider.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * This file is part of Discord4J.
+ *
+ * Discord4J is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Discord4J is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Discord4J. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package discord4j.common;
+
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.resources.ConnectionProvider;
+import reactor.netty.resources.LoopResources;
+
+import java.util.function.Function;
+
+/**
+ * Provides Reactor Netty resources like an {@link reactor.netty.http.client.HttpClient} that can be customized and
+ * reused across the application.
+ * <p>
+ * Allow a user to externally manage the connection pool through a custom
+ * {@link reactor.netty.resources.ConnectionProvider}, and custom event loop threads using a
+ * {@link reactor.netty.resources.LoopResources}.
+ */
+public class ReactorResourceProvider {
```

Comment:
The naming here isn't clear to me. In ClientBuilder it is referred to as the "rest resource provider."

Reply:
+1 for Http